### DISCUSSION TITLE
[Bug]: useFieldArray `append(...)` is reflected in `watch()`, but not in `useWatch()`

### DISCUSSION THREAD

**Discussion by azrosen92** [react-hook-form/react-hook-form#6714] [Q&A] [answered]
### Version Number

6.9.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-watch-forked-5m4tk?file=/src/index.js

### Steps to reproduce

1. Go to https://codesandbox.io/s/react-hook-form-usefieldarray-watch-forked-5m4tk?file=/src/index.js
2. Click on "Append"
3. See console output – there are two `console.log`s, one logging `watch(...)` and one logging `useWatch(...)`
4. The log of `watch` immediately shows the new item in the field array. `useWatch` isn't updated until another character is typed into the new form field or it's unfocused.

### Expected behaviour

I would expect `useWatch` and `watch` to return the same form state at this point.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
<img width="721" alt="Screen Shot 2021-10-05 at 10 27 37 am" src="https://user-images.githubusercontent.com/10513364/135937717-5d0403a2-14dd-46b0-b374-46cab187f15d.png">

can you use the latest v6?

**Reply by azrosen92** [reply]
Upgraded to 6.15.1 and it's working as expected now. Thank you so much!

**Reply by bluebill1049** [reply]
no worries 👍 i would recommend moving to v7 (if you can), it's a lot better with useFieldArray.
